### ISSUE TITLE
Enhancement: Warn about "significantly different" source filename

### ISSUE THREAD

**Issue by stdedos** [koalaman/shellcheck#1867]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC1090
- My shellcheck version (`shellcheck --version` or "online"): 
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
source "$(dirname "$0")/lib/logging.sh"
source "$(dirname "$0")/lib/config.sh"
```

It might be tempting (and go unnoticed), that you fill the first line, and you only duplicate the comment on the second one.

```sh
#!/bin/bash
# shellcheck source=SCRIPTDIR/lib/logging.sh
source "$(dirname "$0")/lib/logging.sh"
# shellcheck source=SCRIPTDIR/lib/logging.sh
source "$(dirname "$0")/lib/config.sh"
```

#### Here's what shellcheck currently says:

Nothing

#### Here's what I wanted or expected to see:

```
# shellcheck source=SCRIPTDIR/lib/logging.sh
source "$(dirname "$0")/lib/config.sh"
       ^-- SC1090: Are you sure that this is the correct source to follow?
```

I know that this can be highly controversial, so, I just want to open the discussion about it.

My idea is that, in python terms, this check would come:

```python
if os.path.basename(directive)[0] not in os.path.basename(source)[0]:
	emit_notice(...)
```
